src/pages/status/index.tsx

```diff
@@ -1,48 +1,170 @@
+      ),
+    },
+    {
+      label: 'Connected to Node API',
+      isOk: status.apiConnection,
+      component: <NodeConnectionCheck isOk={status.apiConnection} />,
+    },
+    {
+      label: 'Connected to Peers',
+      isOk: status.topology,
+      component: <PeerConnection isOk={status.topology} topology={topology} />,
+    },
+  ]
+
+  useEffect(() => {
+    // If the user already changed the active step we don't want to overwrite it
+    if (activeStep >= 0 && activeStep < steps.length) return
+
+    // If any step is not fully loaded yet return
+    if (!isLoading) return
+
+    // Select first step that is not OK
+    // This is deliberately a for loop (and not forEach) so that we can terminate the useEffect from within the cycle
+    for (let i = 0; i < steps.length; ++i) {
+      if (!steps[i].isOk) {
+        setActiveStep(i)
+
+        return
+      }
+    }
+  }, [steps])
+
+  const handleNext = () => {
+    setActiveStep(prevActiveStep => prevActiveStep + 1)
+  }
+
+  const handleBack = () => {
+    setActiveStep(prevActiveStep => prevActiveStep - 1)
+  }
 
   return (
-    <div className={classes.root}>
-      <StatusCard
-        userBeeVersion={latestUserVersion}
-        isLatestBeeVersion={isLatestBeeVersion}
-        isOk={status.all}
-        nodeTopology={topology}
-        latestUrl={latestBeeVersionUrl}
-        nodeAddresses={nodeAddresses}
-      />
-      {nodeAddresses && chequebookAddress && (
-        <EthereumAddressCard nodeAddresses={nodeAddresses} chequebookAddress={chequebookAddress} />
-      )}
-      <NodeSetupWorkflow />
-    </div>
+    <Paper className={classes.root}>
+      <Typography variant="h5" gutterBottom>
+        Node Setup
+      </Typography>
+      <Stepper nonLinear activeStep={activeStep} orientation="vertical">
+        {steps.map(({ label, isOk, component }, index) => (
+          <Step key={label}>
+            <StepLabel
+              onClick={() => setActiveStep(index === activeStep ? steps.length : index)}
+              StepIconComponent={() => {
+                if (isLoading) return <Autorenew style={{ height: '25px', cursor: 'pointer' }} />
+
+                if (isOk) return <CheckCircle style={{ color: '#32c48d', height: '25px', cursor: 'pointer' }} />
+
+                return <Error style={{ color: '#c9201f', height: '25px', cursor: 'pointer' }} />
+              }}
+            >
+              <StepButton
+                disabled={isLoading}
+                onClick={() => setActiveStep(index === activeStep ? steps.length : index)}
+                style={{ justifyContent: 'space-between' }}
+              >
+                <div style={{ display: 'flex' }}>
+                  <div style={{ marginTop: '5px' }}>{label}</div>
+                  <div style={{ marginLeft: '12px' }}>
+                    {index === activeStep ? <ExpandLessSharp /> : <ExpandMoreSharp />}
+                  </div>
+                </div>
```

Comment:
I think there are some refactoring opportunities here - like extracting some layout components, removing inline style, extracting logic. Not a blocker but I think this component has grown a bit too big 🙂

Reply:
I agree and I'll refactor it when I start implementing the new design (page by page).